### ISSUE TITLE
Test suite relies too heavily on dot_parser, fails to test Python API logic

### ISSUE THREAD

**Issue by ferdnyc** [pydot/pydot#385]
#383 demonstrated not only an error in my code, but a deficiency in our current test suite.

Even though there are hundreds of tests run on every pydot commit, the overwhelming majority of them follow the same pattern: Use a `.dot` file as input, load it in using `pydot.graph_from_dot_file()`, and then ensure that the graph image produced by `pydot.Dot.create()` is the same as what's produced by running the original file through `dot -Tjpe`.

That works well for testing the parser, but it falls short of testing the API logic because it won't exercise any code paths that are only accessible via the Python API, and those code paths are growing in number and complexity as we add automatic quoting logic and other user-friendly enhancements that the parser will never take advantage of.

So, things like the `quote_id_if_necessary()` logic will never be triggered by graph data generated in the parser. Any values that must be quoted, it will read as _already quoted_ (they have to be, for the input data to be valid), so the quotes will already be present in its input data.

In other words, when `dot_parser` reads the following:

```gv
graph "graph" {
"Node A";
"Node B";
"Node A" -> "Node B";
}
```

it effectively makes these API calls:

```python
import pydot
g = pydot.Graph('"graph"', graph_type="graph")
g.add_node(pydot.Node('"Node A"'))
g.add_node(pydot.Node('"Node B"'))
g.add_edge(pydot.Edge('"Node A"', '"Node B"'))
```

Now that we've added logic so that these calls are equivalent:

```python
g = pydot.Graph("graph", graph_type="graph")
g.add_node(pydot.Node("Node A"))
g.add_node(pydot.Node("Node B"))
g.add_edge(pydot.Edge("Node A", "Node B"))
```

The only way to properly _test_ THOSE calls is to write them in a test function. The parser can't do it for us.

We need a lot more API-driven tests.

**Comment by lkk7** [comment]
Great idea, the migration shouldn't be too difficult because the number of tests to be rewritten is tolerable, and a lot of them are automatically generated based on the `.dot' files. There are even automated tools that could help (but will probably fail because our tests are messy).

Let me know if you'll be working on the migration and coverage, or if I should do it. It would be bad to duplicate the effort.

**Comment by ferdnyc** [comment]
@lkk7 I'll take a first stab at it, and make it a goal to open a WIP PR early in the process. That way, you & anyone else can pitch in & share your thoughts. (And hopefully, flag me down before I get too far down any wrong paths.)
